src/transcript.rs

```diff
@@ -43,3 +44,85 @@ impl<F: Field> Hasher<F> for DummyHash<F> {
         tmp
     }
 }
+
+/// A transcript that can absorb points from both the base field and scalar
+/// field of a curve
+#[derive(Debug, Clone)]
+pub struct Transcript<C: CurveAffine, HBase, HScalar>
+where
+    HBase: Hasher<C::Base>,
+    HScalar: Hasher<C::Scalar>,
+{
+    // Hasher over the base field
+    base_hasher: HBase,
+    // Hasher over the scalar field
+    scalar_hasher: HScalar,
+    // Indicates if scalar(s) has been hashed but not squeezed
+    scalar_needs_squeezing: bool,
+    // PhantomData
+    _marker: PhantomData<C>,
+}
+
+impl<C: CurveAffine, HBase: Hasher<C::Base>, HScalar: Hasher<C::Scalar>>
+    Transcript<C, HBase, HScalar>
+{
+    /// Initialise a new transcript with Field::one() as keys
+    /// in both the base_hasher and scalar_hasher
+    pub fn new() -> Self {
+        let base_hasher = HBase::init(C::Base::one());
+        let scalar_hasher = HScalar::init(C::Scalar::one());
+        Transcript {
+            base_hasher,
+            scalar_hasher,
+            scalar_needs_squeezing: false,
+            _marker: PhantomData,
+        }
+    }
+
+    /// Initialise a new transcript with some given base_hasher and
+    /// scalar_hasher
+    pub fn init_with_hashers(base_hasher: &HBase, scalar_hasher: &HScalar) -> Self {
```

Comment:
AFAICT this is only used by a test. This isn't an API we should be exposing without some thought as to what guarantees we require from it, so let's keep it internal for now:
```suggestion
    #[cfg(test)]
    pub(crate) fn init_with_hashers(base_hasher: &HBase, scalar_hasher: &HScalar) -> Self {
```